Common/cpp/Tools/BackgroundQueue.h

```diff
@@ -0,0 +1,63 @@
+#pragma once
+
+#include <jsi/jsi.h>
+
+#include "WorkletRuntime.h"
+
+#include <atomic>
+#include <condition_variable>
+#include <memory>
+#include <queue>
+#include <string>
+#include <thread>
+#include <utility>
+#include <vector>
+
+using namespace facebook;
+using namespace react;
+
+namespace reanimated {
+
+class BackgroundQueue : public jsi::HostObject {
+ public:
+  explicit BackgroundQueue(const std::string &name);
+
+  ~BackgroundQueue() override;
+
+  std::string toString() const {
+    return "[BackgroundQueue \"" + name_ + "\"]";
+  }
```

Comment:
More of a programming question than a review bit. Is it a convention/good practice, to make a `toString` method that does not "make sense" as a string? Or is it for Java purposes maybe?

Reply:
This is solely for developer experience purposes so that `String(queue)` returns a nice description of the object. Instead of implementing this directly in `BackgroundQueue::get` method (for `jsi::HostObject`), I moved it to a separate C++ method `BackgroundQueue::toString`.